src/rgw/driver/d4n/rgw_sal_d4n.cc

```diff
@@ -602,9 +602,25 @@ int D4NFilterObject::D4NFilterReadOp::iterate(const DoutPrefixProvider* dpp, int
     ldpp_dout(dpp, 20) << "D4NFilterObject::iterate:: " << __func__ << "(): READ FROM CACHE: oid=" << oid_in_cache << " length to read is: " << len_to_read << " part num: " << start_part_num << 
     " read_ofs: " << read_ofs << " part len: " << part_len << dendl;
 
+    std::string key = oid_in_cache;
```

Comment:
In `flush` method above, for `dirty` objects that start with a prefix `D` you do not pass in the oid that has a prefix `D` and you also do not set the `dirty` flag to true - hence in `update` method, the  `localWeight` will not be updated correctly, because you do that based on adding a prefix `D` after checking for `dirty` flag.

Reply:
We need D_  in case of server failure, so we can recover the dirty data from the local SSD. Otherwise, when the server fails, we will miss all the information about dirty objects.

Reply:
flush() method is used to flush the aio list which comprises of the blocks read from the cache, and since the write workflow has blocks that are dirty, in the read workflow you have already added logic for readin blocks with prefix 'D', the async completion of ios for those blocks in the flush() method needs to be handled correctly.

Reply:
We need to discuss this, I am not sure that I am getting your point correctly.